### PR TITLE
Plugins: Add initial docs

### PR DESCRIPTION
**Pull request by dennisreimann**
[Preview](https://deploy-preview-1208--btcpayserver-docs-preview.netlify.app/Development/Plugins/) - There surely is more to say, but this is a good start and we can extend the information as people request it.

This covers the main points and it seems like people are starting to build plugins, so let's give them a bit of reference and a starting point to get going.

In addition to that I also added the [plugin template repo](https://github.com/btcpayserver/btcpayserver-plugin-template), which we could extend to better cover the more code related parts, so that this doc must not be heavy with code samples.

### CODE REVIEW

File: docs/Development/Plugins.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+# extend search path
+SET search_path TO "BTCPayServer.Plugins.Template", public;
+
+# table list now also shows the template plugin tables
+\dt
+```
+
+### UI Extension Points
+
+The extension poins offer you ways to add your plugin viewa and partials to the UI.
```
Comment (NicolasDorier) [inline docs/Development/Plugins.md:117]: typo

**Comment by NicolasDorier** [comment]
It's a good start, however, how can people debug the plugin?

**Comment by dennisreimann** [comment]
I develop the plugins referenced in the BTCPay solution and debugging works just as in the regular app. Or what do you mean?

File: docs/Development/Plugins.md (hunk 1 above)
Comment (dennisreimann) [reply docs/Development/Plugins.md:117]:
```suggestion
The extension points offer you ways to add your plugin views and partials to the UI.
```

**Comment by grant3345** [comment]
I was scammed and they made a with draw with bit coin and I can't get money back please help. Email is granttheman0702@gmail.com

**Pull request merged**